api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java

```diff
@@ -257,4 +259,17 @@ public boolean isProviderIdentifierUnique(Provider provider) throws APIException
 	public Provider getProviderByIdentifier(String identifier) {
 		return dao.getProviderByIdentifier(identifier);
 	}
+	
+	@Override
+	public Provider createProviderFromUser(User user) {
+		if (user != null) {
```

Comment:
It's better fail out loud if user is null

Reply:
You might want to first check if the user doesn't yet have a provider account

Reply:
I see you make the check below but can you do it earlier please? i.e. before instantiating the Provider object